test/service/FileTransferService.test.ts

```diff
@@ -74,9 +92,84 @@ describe('FileTransferService', () => {
 
             try {
                 await fileTransferService.upload(Buffer.from('This is a test'), 'test.not_supported');
-            } catch(err) {
+            } catch (err) {
                 expect(err.message).to.contain('Unsupported file type');
             }
         });
     });
+
+    describe('File Metadata', () => {
+
+        const fileMetadaUrl = `${URI}/${fileID}`;
+
+        it('should throw an error if the axios request fails', async () => {
+
+            createGetNockRequest(fileMetadaUrl).reply(404);
+
+            try {
+                await fileTransferService.getFileMetadata(fileID);
+            } catch (err) {
+                expect(err.constructor.name).to.eq(FileNotFoundError.name);
+            }
+
+        });
+        it('should get file metada if it exists', async () => {
+
+            createGetNockRequest(fileMetadaUrl).reply(200, expectedMetada);
+
+            await fileTransferService.getFileMetadata(fileID).then(_ => assert.deepEqual(_, expectedMetada));
+        });
+    });
+
+    describe('Download a file', async () => {
+        const readFile = promisify(Fs.readFile);
+
+        const fileToStream = 'package.json';
+        const fileToStreamPath = `./${fileToStream}`;
+        const downloadFileName = 'hello.txt';
+        const downloadedDirPath = `./test`;
+        const downloadedFilePath = `${downloadedDirPath}/${downloadFileName}`;
+        const downloadUrl = `${URI}/${fileID}/download`;
+        const fileDataBuffer = Fs.createReadStream(fileToStream);
```

Comment:
All that reading from file is bit over complicated. I suggest simple `Readable.from(string)`. is similar to how upload test operates on buffer.